arch/arm/include/cortex_m/exc.h

```diff
@@ -48,6 +48,33 @@ static ALWAYS_INLINE bool z_arch_is_in_isr(void)
 	return (__get_IPSR()) ? (true) : (false);
 }
 
+/**
+ * @brief Find out if we were in ISR context
+ *        before the current exception occurred.
+ *
+ * A function that determines, based on inspecting the current
+ * ESF, whether the processor was in handler mode before entering
+ * the current exception state (i.e. nested exception) or not.
+ *
+ * Notes:
+ * - The function shall only be called from ISR context.
+ * - We do not use ARM processor state flags to determine
+ *   whether we are in a nested exception; we rely on the
+ *   RETPSR value stacked on the ESF. Hence, the function
+ *   assumes that the ESF stack frame has a valid RETPSR
+ *   value.
+ *
+ * @param esf the exception stack frame (cannot be NULL)
+ * @return true if execution state was in handler mode, before
+ *              the current exception occurred, otherwise false.
+ */
+static ALWAYS_INLINE bool z_arm_is_in_nested_exception(const z_arch_esf_t *esf)
+{
+	return (esf->basic.xpsr & IPSR_ISR_Msk) ? (true) : (false);
+}
+
+#define z_arch_is_in_nested_exception z_arm_is_in_nested_exception
```

Comment:
don't do this. it's impossible to prototype properly.
define the inline function as z_arch_is_in_nested_exception() if we're going to make this part of the arch interface.

Reply:
Fine, I will change this.

Reply:
Changed now